src/mode/mod.rs

```diff
@@ -31,11 +31,32 @@ where
 {
     /// Clear the display.
     pub fn clear(&mut self) -> Result<(), DisplayError> {
-        self.set_draw_area((0, 0), self.dimensions())?;
+        let old_addr_mode = self.addr_mode;
+        if old_addr_mode != AddrMode::Horizontal {
+            self.set_addr_mode(AddrMode::Horizontal)?;
+        }
 
-        // TODO: If const generics allows this, replace `1024` with computed W x H value for current
-        // `SIZE`.
-        self.draw(&[0u8; 1024])
+        let dim = self.dimensions();
+        self.set_draw_area((0, 0), dim).unwrap();
```

Comment:
This can be an early return on error - no need to panic. Alas no reduction in binary size :(

```suggestion
        self.set_draw_area((0, 0), dim)?;
```